### ISSUE TITLE
Visible Cursor

### ISSUE THREAD

**Issue by abalabahaha** [itdelatrisu/opsu#45]
When a beatmap is loading after you click on it in the song menu, the mouse arrow becomes visible and stays visible until it is moved out of the window and back in.
Affects: OSX 10.10 + jdk1.8.0_31/jdk1.7.0_75

**Comment by itdelatrisu** [comment]
Does this only happen on OSX?  (I don't own any Apple devices, but haven't been able to reproduce this on Windows.)

**Comment by abalabahaha** [comment]
I cannot replicate this in Linux (Ubuntu either). Should be just OSX.

**Comment by itdelatrisu** [comment]
Hm, okay.  I don't know where to start looking, then... and it's really a strange bug if it doesn't happen anywhere else.  I'll mark this as "help wanted", but I really doubt it's a problem with my code.  I didn't look in depth, but it seems like LWJGL has had a number of cursor-related bugs for OSX; these might have been fixed in newer versions...

Let me know if anything else is wrong on OSX, since I haven't run opsu! on a Mac since the initial release. x.x

**Comment by abalabahaha** [comment]
Uncommenting states/Game.java line 842 fixes this, but that may have adverse side effects (can't move mouse out of window, whatever touchscreen issue the line above that talked about)...

**Comment by itdelatrisu** [comment]
And this doesn't happen in any other state?  It doesn't make sense that `setMouseGrabbed()` would fix the issue since it's not being called anywhere else, so I wonder why it works...

I actually had the line there because osu! binds the mouse inside the window during gameplay -- that part is desirable behavior.  But for some reason it also completely disables touchscreen clicks for me, so I can't enable it as-is.

**Comment by abalabahaha** [comment]
I remember it sometimes occurring in other states, but not very often.
I think it might have to do with tiny lag spikes (beatmap load) and the program temporarily relinquishing hold of the mouse... Just a guess.

**Comment by itdelatrisu** [comment]
I guess you could call `setMouseGrabbed(true)` during every state's `enter()` method and then disable it immediately on the first `update()` call (or something like that), but it sounds like an awful hack.

Is it only happening in the game state due to the image loading on the first run?  (e.g. does the cursor appear anytime besides the first game?)  If so, it might not be too bad to grab the mouse before image loading and ungrab it afterwards -- at the beginning and end of the `loadImages()` method.  Could you try it?

**Comment by abalabahaha** [comment]
Nope, it appears every time I enter a beatmap. Grabbing mouse at beginning and and ungrab at end of loadImages didn't work. Sticking the ungrab anywhere in update() didn't work either, the cursor appeared at song start :/
